### ISSUE TITLE
Auto-completion for fish shell

### ISSUE THREAD

**Issue by athityakumar** [athityakumar/colorls#163]
### Description

The auto-completion for flags and paths can be provided for the fish shell via the [getopt](https://github.com/fisherman/getopts) plugin.

- Relevant Issues : #160 
- Relevant PRs : (none)
- Type of issue :
  - [ ] Installation
  - [ ] Font-related
  - [x] Feature request
  - [ ] Bug in existing feature
  - [ ] Developer mode : Code quality / Tests / Documentation

**Comment by dkowis** [comment]
https://fishshell.com/docs/current/commands.html#complete

This is probably a better idea than using the getopt plugin? That way no extra plugin has to be installed, I think.

**Comment by SpaceWhite** [comment]
+1 for this feature

**Comment by adcpe** [comment]
This issue is interesting to me. I agree that adding completions to Fish natively would be better than adding them through an external plugin.

To do that we have to add this feature to Fish directly. I see [PR 179](https://github.com/athityakumar/colorls/pull/179) has a template of what would be added. Is that still good or does it need to be modified to be added to Fish?

**Comment by dkowis** [comment]
I don't think it needs to be added directly to fish. I think you just need to provide the completions file to be loaded.

https://fishshell.com/docs/current/completions.html#completion-own

It's been a long long time since I looked at this tho, heh

**Comment by adcpe** [comment]
It _doesn't_ need to be added Fish, but it would be the best way because it means the user doesn't have to make any effort.

The other way would be add a command to colorls that copies the completions to `$HOME/.config/fish/completions/`.

Any of the two seem cleaner than adding an external plugin like getopt, as suggested.
